src/manifestLayout.js

```diff
@@ -55,14 +49,33 @@ var manifestLayout = function(options) {
         // fixedHeightRows: fixedHeightLine,
         grid: gridAlign
       },
-      viewport = {
-        margins: {},
-        overviewPadding: {},
-        detailPadding: {},
-        width: containerWidth,
-        height: containerHeight,
-        aspectRatio: containerWidth/containerHeight
-      };
+      viewport = viewport(containerWidth, containerHeight, options.viewportPadding);
+
+  function viewport(width, height, viewportPadding) {
+    var horizontalMargin,
+        verticalMargin;
+
+    var viewport = {
+      padding: viewportPadding || {
```

Comment:
Should we validate this at all? eg (top + bottom < 100) .. Just a thought, probably can happen later.